src/condor_startd.V6/Resource.cpp

```diff
@@ -521,14 +521,13 @@ Resource::kill_claim(const std::string& reason, int code, int subcode)
 	case preempting_state:
 			// We might be in preempting/vacating, in which case we'd
 			// still want to do the activity change into killing...
-			// Added 4/26/00 by Derek Wright <wright@cs.wisc.edu>
-		setVacateReason(reason, code, subcode);
-		change_state( preempting_state, killing_act );
 		if (ep_eventlog.inEvent(ULOG_EP_VACATE_CLAIM, this) || ep_eventlog.noEvent()) {
 			auto & ep_event = ep_eventlog.composeEvent(ULOG_EP_VACATE_CLAIM, this);
 			ep_event.Ad().Assign("Kill", true);
 			if (r_cur) r_cur->setVacateInfo(ep_event);
 		}
+		setVacateReason(reason, code, subcode);
+		change_state( preempting_state, killing_act );
```

Comment:
the setVacateReason line can't be moved down without breaking things.   just move the change_state line to after the ep_eventlog stuff, but leave the setVacateReason line alone.